larray/core/array.py

```diff
@@ -1374,71 +1376,104 @@ def reindex(self, axes_to_reindex=None, new_axis=None, fill_value=nan, inplace=F
         >>> arr = ndtest((2, 2))
         >>> arr
-        a\\b  b0  b1
+        a\b  b0  b1
          a0   0   1
          a1   2   3
         >>> arr2 = ndtest('a=a1,a2;c=c0;b=b2..b0')
         >>> arr2
-         a  c\\b  b2  b1  b0
+         a  c\b  b2  b1  b0
         a1   c0   0   1   2
         a2   c0   3   4   5
 
         Reindex an axis by passing labels (list or string)
 
         >>> arr.reindex('b', ['b1', 'b2', 'b0'])
-        a\\b   b1   b2   b0
+        a\b   b1   b2   b0
          a0  1.0  nan  0.0
          a1  3.0  nan  2.0
         >>> arr.reindex('b', 'b0..b2', fill_value=-1)
-        a\\b  b0  b1  b2
+        a\b  b0  b1  b2
+         a0   0   1  -1
+         a1   2   3  -1
+        >>> arr.reindex(b='b=b0..b2', fill_value=-1)
+        a\b  b0  b1  b2
          a0   0   1  -1
          a1   2   3  -1
 
         Reindex using an axis from another array
 
         >>> arr.reindex('b', arr2.b, fill_value=-1)
-        a\\b  b2  b1  b0
+        a\b  b2  b1  b0
          a0  -1   1   0
          a1  -1   3   2
 
         Reindex using a subset of an axis
 
         >>> arr.reindex('b', arr2.b['b1':], fill_value=-1)
-        a\\b  b1  b0
+        a\b  b1  b0
+         a0   1   0
+         a1   3   2
+
+        Reindex by passing an axis or a group
+
+        >>> arr.reindex('b=b2..b0', fill_value=-1)
+        a\b  b2  b1  b0
+         a0  -1   1   0
+         a1  -1   3   2
+        >>> arr.reindex(arr2.b, fill_value=-1)
+        a\b  b2  b1  b0
+         a0  -1   1   0
+         a1  -1   3   2
+        >>> arr.reindex(arr2.b['b1':], fill_value=-1)
+        a\b  b1  b0
          a0   1   0
          a1   3   2
 
         Reindex several axes
 
         >>> arr.reindex({'a': arr2.a, 'b': arr2.b}, fill_value=-1)
-        a\\b  b2  b1  b0
+        a\b  b2  b1  b0
          a1  -1   3   2
          a2  -1  -1  -1
         >>> arr.reindex({'a': arr2.a, 'b': arr2.b['b1':]}, fill_value=-1)
-        a\\b  b1  b0
+        a\b  b1  b0
          a1   3   2
          a2  -1  -1
+        >>> arr.reindex(a=arr2.a, b=arr2.b, fill_value=-1)
+        a\b  b2  b1  b0
+         a1  -1   3   2
+         a2  -1  -1  -1
 
         Reindex by passing a collection of axes
 
         >>> arr.reindex(arr2.axes, fill_value=-1)
-         a  b\\c  c0
+         a  b\c  c0
         a1   b2  -1
         a1   b1   3
         a1   b0   2
         a2   b2  -1
         a2   b1  -1
         a2   b0  -1
         >>> arr2.reindex(arr.axes, fill_value=-1)
-         a  c\\b  b0  b1
+         a  c\b  b0  b1
         a0   c0  -1  -1
         a1   c0   2   1
         """
         # XXX: can't we move this to AxisCollection.replace?
+        if isinstance(axes_to_reindex, basestring):
+            try:
+                axes_to_reindex = Axis(axes_to_reindex)
+            except:
+                pass
+        if isinstance(axes_to_reindex, (Group, Axis)):
+            new_axis = axes_to_reindex.copy() if isinstance(axes_to_reindex, Axis) else Axis(axes_to_reindex)
```

Comment:
why do you need the copy() here? I guess there is a good reason, but since I fail to see it, adding a comment wouldn't hurt 😉

Reply:
I tried something different at first and I forgot to remove the .copy()